### DISCUSSION TITLE
What events are available from onAuthStateChange?

### DISCUSSION THREAD

**Discussion by horaceho** [supabase/supabase#3031] [Questions] [answered]
A `SIGNED_IN` event is fired by `supabase.auth.onAuthStateChange((event, session)` after a successful login. Are there other events, such as `SIGNED_OUT`? https://supabase.io/docs/reference/javascript/auth-onauthstatechange does not mention this.

**Comment by silentworks** [comment, marked as answer]
Here is a list of all the ones I could find:

- SIGNED_IN
- SIGNED_OUT
- USER_UPDATED
- USER_DELETED
- PASSWORD_RECOVERY

You can see these in the gotrue-js source code https://github.com/supabase/gotrue-js/blob/master/src/lib/types.ts#L13-L18
